src/testExplorer/util.ts

```diff
@@ -64,19 +65,60 @@ export function gatherTestItems(
 }
 
 /**
- * Return prefixes of fully qualified name.
+ * @member id - fully qualified name of testItem
+ * @member label - last part of fully qualified name
+ * @member next - returns next prefix, if it exists.
+ * If fully qualified named was traversed (last part was reached) returns null
+ *
+ * next calls for 'a.b.c.d.TestSuite' will yield
+ * - {id: a, label: a}
+ * - {id: a.b, label: b}
+ * - {id: a.b.c, label: c}
+ * - {id: a.b.c.d, label: d}
+ * - {id: a.b.c.d.TestSuite, label: TestSuite} (if prefixesOf was called with includeSelf = true)
+ * - null
+ */
+export interface TestItemPath {
+  readonly id: string;
+  readonly label: string;
+  readonly next: () => TestItemPath | null;
+}
+
+/**
+ * Generate prefixes of fully qualified name for test items
  * includeSelf = false :
  * 'a.b.c.d.TestSuite' -> ['a', 'a.b', 'a.b.c', 'a.b.c.d']
  * includeSelf = true :
  * 'a.b.c.d.TestSuite' -> ['a', 'a.b', 'a.b.c', 'a.b.c.d', 'a.b.c.d.TestSuite']
  */
-export function prefixesOf(str: string, includeSelf = false): string[] {
-  const parts = str.split(".");
-  const prefixes = parts
-    .map((_, idx) => {
-      const joined = parts.slice(0, idx).join(".");
-      return joined;
-    })
-    .filter((p) => p.length > 0);
-  return includeSelf ? [...prefixes, str] : prefixes;
+export function prefixesOf(
+  fullyQualifiedName: FullyQualifiedClassName,
+  includeSelf = false
+): TestItemPath | null {
+  const splitted = fullyQualifiedName.split(".");
```

Comment:
```suggestion
  const splitString = fullyQualifiedName.split(".");
```
Maybe? I hurts me a bit to see `splitted` :sweat_smile: